diff --git a/ortools/constraint_solver/routing_filter_committables.h b/ortools/constraint_solver/routing_filter_committables.h
index 6883f65..bd21f2a 100644
--- a/ortools/constraint_solver/routing_filter_committables.h
+++ b/ortools/constraint_solver/routing_filter_committables.h
@@ -58,7 +58,7 @@
 class CommittableArray {
  public:
   // Makes a vector with initial elements all committed to value.
-  CommittableArray<T>(size_t num_elements, const T& value)
+  CommittableArray(size_t num_elements, const T& value)
       : elements_(num_elements, {value, value}), changed_(num_elements) {}
 
   // Return the size of the vector.
