diff --git a/setup.py b/setup.py
index 39ca17e..7f829d0 100644
--- a/setup.py
+++ b/setup.py
@@ -330,13 +330,13 @@ if BUILD_WITH_SYSTEM_CARES:
 
 if BUILD_WITH_SYSTEM_RE2:
     CORE_C_FILES = filter(lambda x: "third_party/re2" not in x, CORE_C_FILES)
-    RE2_INCLUDE = (os.path.join("/usr", "include", "re2"),)
+    RE2_INCLUDE =  tuple(os.environ["GRPC_PYTHON_RE2_INCLUDE"].split(":")) if "GRPC_PYTHON_RE2_INCLUDE" in os.environ else (os.path.join("/usr", "include", "re2"),)
 
 if BUILD_WITH_SYSTEM_ABSL:
     CORE_C_FILES = filter(
         lambda x: "third_party/abseil-cpp" not in x, CORE_C_FILES
     )
-    ABSL_INCLUDE = (os.path.join("/usr", "include"),)
+    ABSL_INCLUDE =  tuple(os.environ["GRPC_PYTHON_ABSL_INCLUDE"].split(":")) if "GRPC_PYTHON_ABSL_INCLUDE" in os.environ else (os.path.join("/usr", "include"),)
 
 EXTENSION_INCLUDE_DIRECTORIES = (
     (PYTHON_STEM,)
@@ -380,9 +380,17 @@ if BUILD_WITH_SYSTEM_RE2:
 if BUILD_WITH_SYSTEM_ABSL:
     EXTENSION_LIBRARIES += tuple(
         lib.stem[3:]
-        for lib in sorted(pathlib.Path("/usr").glob("lib*/libabsl_*.so"))
+        for lib in sorted(pathlib.Path(os.environ["GRPC_PYTHON_ABSL_LIB_DIR"]).glob("libabsl_*.so") if "GRPC_PYTHON_ABSL_LIB_DIR" in os.environ else pathlib.Path("/usr").glob("lib*/libabsl_*.so"))
     )
 
+EXTENSION_LIBRARY_DIRECTORIES = ()
+if BUILD_WITH_SYSTEM_ABSL:
+    if os.environ.get("GRPC_PYTHON_ABSL_LIB_DIR"):
+        EXTENSION_LIBRARY_DIRECTORIES += (os.environ["GRPC_PYTHON_ABSL_LIB_DIR"],)
+if BUILD_WITH_SYSTEM_RE2:
+    if os.environ.get("GRPC_PYTHON_RE2_LIB_DIR"):
+        EXTENSION_LIBRARY_DIRECTORIES += (os.environ["GRPC_PYTHON_RE2_LIB_DIR"],)
+
 DEFINE_MACROS = (("_WIN32_WINNT", 0x600),)
 asm_files = []
 
@@ -507,6 +515,7 @@ def cython_extensions_and_necessity():
             ),
             include_dirs=list(EXTENSION_INCLUDE_DIRECTORIES),
             libraries=list(EXTENSION_LIBRARIES),
+            library_dirs=list(EXTENSION_LIBRARY_DIRECTORIES),
             define_macros=list(DEFINE_MACROS),
             extra_objects=extra_objects,
             extra_compile_args=list(CFLAGS),
