--- a/intrinsic/platform/pubsub/zenoh_util/zenoh_helpers.cc
+++ b/intrinsic/platform/pubsub/zenoh_util/zenoh_helpers.cc
@@ -2,7 +2,9 @@
 
 #include "intrinsic/platform/pubsub/zenoh_util/zenoh_helpers.h"
 
+#include <array>
 #include <cstdlib>
+#include <filesystem>
 #include <string>
 #include <vector>
 
@@ -13,12 +15,12 @@
 #include "absl/strings/str_split.h"
 #include "absl/strings/string_view.h"
 #include "ortools/base/file.h"
-#include "ortools/base/path.h"
-#include "rules_cc/cc/runfiles/runfiles.h"
+// #include "ortools/base/path.h"
+// #include "rules_cc/cc/runfiles/runfiles.h"
 
 namespace intrinsic {
 
-using ::rules_cc::cc::runfiles::Runfiles;
+// using ::rules_cc::cc::runfiles::Runfiles;
 
 bool RunningUnderTest() {
   return (getenv("TEST_TMPDIR") != nullptr) ||
@@ -29,33 +31,32 @@
   return getenv("KUBERNETES_SERVICE_HOST") != nullptr;
 }
 
+#ifndef INTRINSIC_SDK_CMAKE_LIB_PATH
+#error "Compiler definition INTRINSIC_SDK_CMAKE_LIB_PATH is required."
+#endif
+
+#ifndef INTRINSIC_SDK_CMAKE_SHARE_PATH
+#error "Compiler definition INTRINSIC_SDK_CMAKE_SHARE_PATH is required."
+#endif
+
 std::string GetZenohRunfilesPath(absl::string_view file_path) {
-  // The simplest scenario is that the file is available at a global
-  // path, which happens when running in Kubernetes or in Docker.
-  std::string global_path = absl::StrCat("/", file_path);
-  if (file::Exists(global_path, file::Defaults()).ok()) {
-    return global_path;
+  // Patched by intrinsic-dev/intrinsic_sdk_ros.git to work in a CMake context.
+  // TODO(wjwwood): consider using ament_index to make this code relocatable.
+  std::array<std::filesystem::path, 2> prefix_paths = {
+    std::filesystem::path(INTRINSIC_SDK_CMAKE_LIB_PATH),
+    std::filesystem::path(INTRINSIC_SDK_CMAKE_SHARE_PATH)
+  };
+
+  // Prefer a more complete path in the lib or share path over just the give file_path.
+  for (const auto & prefix_path : prefix_paths) {
+    std::filesystem::path possible_path = prefix_path / file_path;
+    if (std::filesystem::exists(possible_path)) {
+      return possible_path.string();
+    }
   }
 
-  // Otherwise, try to calculate a path using Bazel runfiles.
-  std::string error;
-  std::string repository = BAZEL_CURRENT_REPOSITORY;
-  std::string apparentRepoName = "";
-  std::unique_ptr<Runfiles> runfiles;
-  if (RunningUnderTest()) {
-    runfiles.reset(Runfiles::CreateForTest(repository, &error));
-  } else {
-    runfiles.reset(Runfiles::Create(program_invocation_name, &error));
-  }
-  if (!runfiles) {
-    LOG(FATAL) << "Could not initialize runfiles: " << error;
-  }
-  if (RunningInKubernetes() || RunningUnderTest()) {
-    apparentRepoName = "ai_intrinsic_sdks";
-  } else {
-    apparentRepoName = repository;
-  }
-  return runfiles->Rlocation(absl::StrCat(apparentRepoName, "/", file_path));
+  // Return the original file_path if nothing better was found.
+  return std::string(file_path);
 }
 
 absl::Status ValidZenohKeyexpr(absl::string_view keyexpr) {
