--- a/incode/middleware/zenoh/BUILD
+++ b/incode/middleware/zenoh/BUILD
@@ -24,6 +24,7 @@
         "imw_zenoh_subscription.h",
     ],
     linkstatic = True,
+    alwayslink = True,
     visibility = ["//visibility:public"],
     deps = [
         "//incode/middleware:imw_interface",
@@ -40,3 +41,18 @@
     "config/zenoh_router_config.json5",
     "config/zenoh_peer_config.json5",
 ])
+
+cc_binary(
+    name = "libimw_zenoh.so",
+    linkshared = True,
+    linkopts = [
+        "-Wl,-soname,libimw_zenoh.so",
+        "-Wl,--version-script,$(location imw-versionscript.lds)",
+    ],
+    additional_linker_inputs = ["imw-versionscript.lds"],
+    visibility = ["//visibility:public"],
+    deps = [
+        ":imw_zenoh",
+    ],
+)
+
--- a/incode/middleware/zenoh/imw-versionscript.lds
+++ b/incode/middleware/zenoh/imw-versionscript.lds
@@ -3,5 +3,6 @@
 {
   global:
     imw_*;
+    _ZN9intrinsic*;
   local: *;
 };
